### ISSUE TITLE
Boolean option for RadioButtonGroup

### ISSUE THREAD

**Issue by smeet2296** [dohomi/react-hook-form-mui#294]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Currently RadioButtonGroup Element supports string and number. But if I want to use a Yes/No radio button which should store true/false its not working.

Example optionsArray : [{"id":true,"value":"Yes"},{"id":false,"value":"No"}]

If for reducing complexity there is a way that the options have string true/false values and then the value gets stored as boolean that would also be fine

**Comment by dohomi** [comment]
any PR is welcome

**Comment by smeet2296** [comment]
I can create PR tomorrow. How long does it take to get merged and released?

**Comment by smeet2296** [comment]
@dohomi 

I found a workaround

For the example optionsArray mentioned above, we can do the following

> transform={{
        input: (value) => {
          return value != null ? value : "";
        },
        output: (_, value) => {
          return value === "true" ? true : value === "false" ? false : "";
        },
      }}
      
 The only problem is [https://github.com/react-hook-form/react-hook-form/issues/2323](url)
 
 For this, we can avoid using required in rule as well as prop and use validate function in the rule instead. We will have to add the asterisk in the label manually.

I wanted to add boolean type inbuilt but then due to the above mentioned git issue, required rule and prop will stop working

**Comment by dohomi** [comment]
I really think you make yourself too much trouble: either use a checkbox / switch which handle true/false natively or use "0" "1" and just cast the value shortly before you persist it in the database inside your `handleSubmit`. I am currently loaded with work that I won't spend time on the issue. If you find something usable I am happy to review a PR

**Comment by smeet2296** [comment]
Yes we can do that if there is only a form that we are building. But I am building a form configurator and I cannot have such restrictions that a user cannot configure boolean values through it. The above solution works for now.
